tests/PSRule.Rules.Azure.Tests/Azure.Template.Tests.ps1

```diff
@@ -655,5 +655,39 @@ Describe 'Azure.Template' -Tag 'Template' {
             $targetNames = $ruleResult | ForEach-Object { $_.TargetName.Split([char[]]@('\', '/'))[-1] };
             $targetNames | Should -BeIn 'Resources.Parameters.json', 'Resources.VPN.Parameters3.json';
         }
+
+        It 'Azure.Template.UseComments' {
+            $dataPath = Join-Path -Path $here -ChildPath 'Resources.Template*.json';
+            $result = Invoke-PSRule @invokeParams -InputPath $dataPath -Format None -Name 'Azure.Template.UseComments';
+            $filteredResult = $result | Where-Object { $_.RuleName -eq 'Azure.Template.UseComments' };
+
+            # Fail
+            $ruleResult = @($filteredResult | Where-Object { $_.Outcome -eq 'Fail' });
+            $ruleResult | Should -Not -BeNullOrEmpty;
+            $ruleResult.Length | Should -Be 3;
+            $ruleResult.TargetName | Should -BeLike "*Resources.Template[2-4].json";
+
+            $ruleResult[0].Reason | Should -HaveCount 2;
+            $ruleResult[0].Reason[0] | Should -BeExactly "The field 'comments' does not exist.";
```

Comment:
Was seeing if we could put resource name in the reasons, but this doesn't work out when we use ARM template expressions like `[concat(parameters('vnetName'), '/', parameters('clusterSubnet'), '/Microsoft.Authorization/', guid(parameters('clusterSubnet')))]`.